crates/libs/starknet/types/src/assets.rs

```diff
@@ -30,25 +33,34 @@ impl Asset {
         match self {
             Asset::Strk => "strk",
             Asset::Eth => "eth",
+            Asset::Btc => "btc",
+            Asset::Usdc => "usdc",
+            Asset::Usdt => "usdt",
         }
     }
 
     pub fn precision(&self) -> u8 {
         match self {
             Asset::Strk | Asset::Eth => 18,
+            Asset::Btc => 8,
+            Asset::Usdc | Asset::Usdt => 6,
         }
     }
 
     pub fn scale_factor(&self) -> U256 {
         match self {
             Asset::Strk | Asset::Eth => U256::from(1_000_000_000_000_000_000u64),
+            Asset::Btc => U256::from(100_000_000u64),
+            Asset::Usdc | Asset::Usdt => U256::from(1_000_000u64),
         }
     }
 
     pub fn find_best_unit(&self) -> Unit {
         match self {
             Asset::Strk => Unit::MilliStrk,
             Asset::Eth => Unit::Gwei,
+            Asset::Btc => Unit::Satoshi,
+            Asset::Usdc | Asset::Usdt => Unit::MicroUsd,
```

Comment:
Won't the best unit be the `cent`? Like a 100th of a dollar.
I don't know of any purchases that need more precision than this.
Wdyt?

Reply:
Yes, that makes more sense. I've updated it